CacheManager.StackExchange.Redis.SelfCleaning/SelfCleaningRedisConfigurationBuilderExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using CacheManager.Core;
+using CacheManager.Redis;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning
+{
+    public static class SelfCleaningRedisConfigurationBuilderExtensions
+    {
+        private const string DEFAULT_CONFIGURATION_KEY = "redis";
+        private const string CONFIGURATION_KEY_SUFFIX = "-self_cleaning";
+        
+        public static ConfigurationBuilderCachePart WithSelfCleaningRedisConfiguration(
```

Comment:
Change the order of the functions. Instead of 1, 2, 3, 4 Do 3, 2, 1, 4, because there are two flows. 
Morever, is there any case in which someone calls directly to function 1 or 2 (and not via 3)? If not, so why not making them private?

Reply:
Changed the order. About making these private - I agree about the first (now 3rd), but I don't know about the second. I wonder if there would be any reason to inject the `IConnectionMultiplexer` and/or `ITimer`. I also wonder if instantiating the `ITimer` is good idea in the first place. What do you think?

Reply:
> Oh, I must have missed it. Yes, creating the Timer by yourself isnt a good solution, Or maybe letting the first method stay while the ability to inject different TImer (and IConnectionMulpitlexer) is being reserved using the second method... I think that If we choose to leave the first method, we should change its name to `WithSelfCleaningRedisDefaultConfiguration` or `WithDefaultSelfCleaningRedisConfiguration`, and clearify it in short summary of the function. That way there will be 3 flows in the class-
1. Defualt configuration - intialize everything including some Timer and ConnectionMultiplexter, which use flow 2:
2. Custom configuration in which you control those interfaces
3. the mystic reflection bla bla bla

I think that good enough (although creating some help-methods that use specific implementations isnt conventional in our team).

The only thing to be afraid is that no one will know that he has the ability to change those implementations (by using the second flow directly), and that sucks. But, in order to understand how he will be able to do that, he will have to get into the code, and then he will understand that he can use the second flow... SO I THINK ITS REALLY FINE, JUST GOOD SUMMARIES TO CLASS AND FUNCTIONS AND WE WILL BE OK

מצטער על החפירה אמאלה כמה כתבתי

Reply:
Added summaries for these methods